src/event.rs

```diff
@@ -0,0 +1,616 @@
+				let min = time_forwardable.as_millis() as u64;
+
+				locked_runtime.as_ref().unwrap().spawn(async move {
+					let millis_to_sleep = thread_rng().gen_range(min..min * 5) as u64;
+					tokio::time::sleep(Duration::from_millis(millis_to_sleep)).await;
+
+					forwarding_channel_manager.process_pending_htlc_forwards();
+				});
+			}
+			LdkEvent::SpendableOutputs { outputs } => {
+				// TODO: We should eventually remember the outputs and supply them to the wallet's coin selection, once BDK allows us to do so.
+				let destination_address = self.wallet.get_new_address().unwrap();
+				let output_descriptors = &outputs.iter().map(|a| a).collect::<Vec<_>>();
+				let tx_feerate =
+					self.wallet.get_est_sat_per_1000_weight(ConfirmationTarget::Normal);
+				let spending_tx = self
+					.keys_manager
+					.spend_spendable_outputs(
+						output_descriptors,
+						Vec::new(),
+						destination_address.script_pubkey(),
+						tx_feerate,
+						&Secp256k1::new(),
+					)
+					.unwrap();
+				self.wallet.broadcast_transaction(&spending_tx);
+			}
+			LdkEvent::OpenChannelRequest { .. } => {}
+			LdkEvent::PaymentForwarded {
+				prev_channel_id,
+				next_channel_id,
+				fee_earned_msat,
+				claim_from_onchain_tx,
+			} => {
+				let read_only_network_graph = self.network_graph.read_only();
+				let nodes = read_only_network_graph.nodes();
+				let channels = self.channel_manager.list_channels();
+
+				let node_str = |channel_id: &Option<[u8; 32]>| match channel_id {
+					None => String::new(),
+					Some(channel_id) => match channels.iter().find(|c| c.channel_id == *channel_id)
+					{
+						None => String::new(),
+						Some(channel) => {
+							match nodes.get(&NodeId::from_pubkey(&channel.counterparty.node_id)) {
+								None => "private node".to_string(),
+								Some(node) => match &node.announcement_info {
+									None => "unnamed node".to_string(),
+									Some(announcement) => {
+										format!("node {}", announcement.alias)
+									}
+								},
+							}
+						}
+					},
+				};
+				let channel_str = |channel_id: &Option<[u8; 32]>| {
+					channel_id
+						.map(|channel_id| {
+							format!(" with channel {}", hex_utils::to_string(&channel_id))
+						})
+						.unwrap_or_default()
+				};
+				let from_prev_str = format!(
+					" from {}{}",
+					node_str(&prev_channel_id),
+					channel_str(&prev_channel_id)
+				);
+				let to_next_str =
+					format!(" to {}{}", node_str(&next_channel_id), channel_str(&next_channel_id));
+
+				let from_onchain_str = if claim_from_onchain_tx {
+					"from onchain downstream claim"
+				} else {
+					"from HTLC fulfill message"
+				};
+				if let Some(fee_earned) = fee_earned_msat {
+					log_info!(
+						self.logger,
+						"Forwarded payment{}{}, earning {} msat {}",
+						from_prev_str,
+						to_next_str,
+						fee_earned,
+						from_onchain_str
+					);
+				} else {
+					log_info!(
+						self.logger,
+						"Forwarded payment{}{}, claiming onchain {}",
```

Comment:
Why is "claiming onchain" determined by `fee_earned_msat`?

Reply:
Reworded.